### ISSUE TITLE
Unable to install latest azure-cli-2.0.77-1.el7.x86_64 on RedHat 7

### ISSUE THREAD

**Issue by captainbrown** [Azure/azure-cli#11418]
azure-cli-2.0.77-1.el7.x86_64

**Describe the bug**
The python3 package is not available on RedHat 7, but the latest release of azure-cli-2.0.77-1.el7 requires that to be installed.  This is not available in RedHat 7 repos

**To Reproduce**
$ sudo yum install azure-cli
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-
              : manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
Resolving Dependencies
--> Running transaction check
---> Package azure-cli.x86_64 0:2.0.76-1.el7 will be updated
---> Package azure-cli.x86_64 0:2.0.77-1.el7 will be an update
--> Processing Dependency: python3 for package: azure-cli-2.0.77-1.el7.x86_64
--> Processing Dependency: libpython3.6m.so.1.0()(64bit) for package: azure-cli-2.0.77-1.el7.x86_64
--> Finished Dependency Resolution
Error: Package: azure-cli-2.0.77-1.el7.x86_64 (azure-cli)
           Requires: python3
Error: Package: azure-cli-2.0.77-1.el7.x86_64 (azure-cli)
           Requires: libpython3.6m.so.1.0()(64bit)
 You could try using --skip-broken to work around the problem

**Expected behavior**
I expect it to be able to install the latest version of Azure CLI on an OS that does not have a package for python3 natively

**Environment summary**
NAME="Red Hat Enterprise Linux Server"
VERSION="7.6 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.6"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.6 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.6:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

$ python3 --version
Python 3.4.10

**Comment by captainbrown** [comment]
This was forewarned in https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-yum?view=azure-cli-latest

Closing

**Comment by captainbrown** [comment]
Coming back to this one.  Even if I have python3 installed via the only way to get it on RedHat7, I still cannot install the latest Azure CLI.  I think it's because yum doesn't have a record of the install because it's simply not available in the repository for RedHat 7.  

Does this mean that RedHat 7 is no longer supported for Azure CLI, or am I missing something?

$ python3 --version
Python 3.7.4
$ sudo yum install azure-cli
Loaded plugins: langpacks, product-id, search-disabled-repos
Resolving Dependencies
--> Running transaction check
---> Package azure-cli.x86_64 0:2.0.77-1.el7 will be installed
--> Processing Dependency: python3 for package: azure-cli-2.0.77-1.el7.x86_64
--> Processing Dependency: libpython3.6m.so.1.0()(64bit) for package: azure-cli-2.0.77-1.el7.x86_64
--> Finished Dependency Resolution
Error: Package: azure-cli-2.0.77-1.el7.x86_64 (azure-cli)
           Requires: python3
Error: Package: azure-cli-2.0.77-1.el7.x86_64 (azure-cli)
           Requires: libpython3.6m.so.1.0()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by captainbrown** [comment]
Redhat 8 doesn't seem to resolve this issue either, thanks to the issue I found here:  https://github.com/microsoft/azure-pipelines-agent/issues/2349

So, I suppose Redhat is just unsupported for now?

**Comment by yonzhan** [comment]
@fengzhou-msft  please take a look.

**Comment by fengzhou-msft** [comment]
@captainbrown 2.0.77 should work with RHEL 7.7 and 8. If you're on lower versions, after installing Python 3 which you already did, you can download azure-cli package, and install without dependency.
```
$ sudo yum install yum-utils
$ sudo yumdownloader azure-cli
$ sudo rpm -ivh --nodeps azure-cli-2.0.77-1.el7.x86_64.rpm
```

**Comment by yonzhan** [comment]
Can we close this issue?

**Comment by captainbrown** [comment]
It doesn't seem that my method of installing python3 on RHEL 7 is effective.  This is on RHEL 7.7 after installing rh-python36 and running your suggested commands.

$ az --help
/bin/az: line 2: python3: command not found

**Comment by captainbrown** [comment]
This issue seems to be resolved for RHEL7 for the latest version of Azure CLI.  azure-cli.x86_64 0:2.0.77-1.el7

**Comment by pmozbert** [comment]
amazing how MSFT keeps closing these tickets as resolved, when they are not resolved.

**Comment by fengzhou-msft** [comment]
> amazing how MSFT keeps closing these tickets as resolved, when they are not resolved.

This issue is closed by the submitter. 

Please try with the latest azure-cli version, if your system is not on the [tested list](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-yum?view=azure-cli-latest), please check our troubleshooting [doc](https://github.com/Azure/azure-cli/blob/dev/doc/install_troubleshooting.md#install-on-rhel-76-or-other-yum-managed-systems-without-python-3) for workaround. 

Feel free to submit a new issue if they do not work for you.

**Comment by amchaney** [comment]
Had to do this:
```
$ sudo yumdownloader azure-cli
$ sudo rpm -ivh --nodeps azure-cli-2.0.77-1.el7.x86_64.rpm
```
on Amazon Linux 2 to install since I kept getting an error about a python3 library that couldn't be loaded (even though python 3 was installed).

**Comment by jatinmaya** [comment]
sudo yumdownloader azure-cli
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
azure-cli-2.14.0-1.el7.x86_64.rpm               

sudo rpm -ivh --nodeps azure-cli-2.14.0-1.el7.x86_64.rpm

az version
{
  "azure-cli": "2.13.0",
  "azure-cli-core": "2.13.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {}
}

I am still in version 2.13 @ fengzhou-msft could you please explain this?

**Comment by Kaitou786** [comment]
Just putting it out there if you are using `rpm --nodeps` than it would be worth noting, instead of defining the exact rpm name you can do 

```
sudo rpm -Uvh --nodeps --force $(repoquery --location azure-cli)
```
The force option helps if you are running it CI and the same agent might be used for multiple runs
